lib/list.gi

```diff
@@ -646,6 +646,18 @@ InstallOtherMethod(AsSSortedList,
      [ IsList ],
      l -> AsSSortedListList( AsPlist( l ) ) );
 
+InstallMethod( AsSSortedList,
+    "for a mutable strictly sorted list",
+    [ IsSSortedList and IsMutable ],
+    SUM_FLAGS,
+    Immutable );
+
+InstallMethod( AsSSortedList,
```

Comment:
I understand why we might have this optimisation (ID_FUNC), but Immutable is fast (and doesn't copy) objects which are already immutable, and I never like functions which require another function to be correct.

Reply:
Also, maybe chuck in some more tests (I'd be tempted to chuck in a sorted list of booleans, and a sorted string) as they are "non-standard" list types. Probably not required in this case, but I personally use them whenever I'm sorting lists.